syntax = "proto3";

package userservice;

service UserService {
  // 사용자 정보 조회
  rpc GetUser (UserRequest) returns (UserResponse) {}
  
  // 사용자 목록 조회
  rpc ListUsers (ListUsersRequest) returns (ListUsersResponse) {}
  
  // 사용자 생성
  rpc CreateUser (CreateUserRequest) returns (UserResponse) {}
}

message UserRequest {
  int32 user_id = 1;
}

message UserResponse {
  int32 user_id = 1;
  string name = 2;
  string email = 3;
}

message ListUsersRequest {
  int32 page = 1;
  int32 page_size = 2;
}

message ListUsersResponse {
  repeated UserResponse users = 1;
  int32 total_count = 2;
}

message CreateUserRequest {
  string name = 1;
  string email = 2;
}